File: 3532/solution.java
class Solution {
    public boolean[] pathExistenceQueries(int n, int[] nums, int maxDiff, int[][] queries) {
        int[] clusters = new int[n];
        int cluster = 1;
        boolean[] res = new boolean[queries.length];
        clusters[0] = cluster;
        for(int i=1;i<n;i++){
            if(nums[i]-nums[i-1]>maxDiff)
                cluster++;
            clusters[i] = cluster;
        }
        for(int i=0;i<queries.length;i++){
            int u = queries[i][0];
            int v = queries[i][1];
            if(clusters[u]==clusters[v] || u==v)
                res[i] = true;
            else
                res[i] = false;
        } 
        return res;
    }
}

File: 3532/solution.py
class Solution:
    def pathExistenceQueries(self, n: int, nums: List[int], maxDiff: int, queries: List[List[int]]) -> List[bool]:
        clusterNo = 1
        clusters = [1]
        res = []
        for i in range(1,n):
            if nums[i]-nums[i-1]>maxDiff:
                clusterNo+=1
            clusters.append(clusterNo)
        for u,v in queries:
            if clusters[u]==clusters[v] or u==v:
                res.append(True)
            else:
                res.append(False)
        return res
        

